app/src/main/java/com/github/se/travelpouch/model/notifications/NotificationRepositoryFirestore.kt

```diff
@@ -0,0 +1,153 @@
+package com.github.se.travelpouch.model.notifications
+
+import android.util.Log
+import com.github.se.travelpouch.model.travels.Role
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.Query
+
+/**
+ * Repository class for managing notifications in the Firestore database.
+ *
+ * @property firestore The Firestore instance used for database operations.
+ */
+class NotificationRepositoryFirestore(private val firestore: FirebaseFirestore) :
+    NotificationRepository {
+
+  // Reference to the "notifications" collection in Firestore
+  private val notificationCollection = firestore.collection("notifications")
```

Comment:
I changed it as
`private val notificationCollection = firestore.collection(FirebasePaths.notifications)`

I keep the variable notificationCollection to avoid duplication. In other repository they call each time the collection on the database, ex. `db.collection(userCollectionPath)`

Reply:
Yes, this how it needs to be done. Keeping it as you are doing it will cause a Firebase Internal Error. Please follow my comment